brave/brave-core
# [Brave News]: Separate PrefService from BraveNews* classes

Resolves https://github.com/brave/brave-browser/issues/36558

This should make it easier to push the majority of the Brave News services into a different thread.

Sorry for the monster PR, everything was quite tightly coupled. Additionally, a lot of the new code is tests.



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

There should be no user facing changes as a result of this refactor. I think the most likely place for bugs to have snuck in will be in the observations. To that end:

1. Subscribe to lots of publishers as you can, as quickly as you can.
2. Restart Brave
3. The list of Publishers should not have changed
4. Subscribe to lots of channels as quick as you can
5. Restart Brave
6. The list of Channels should not have changed
7. Subscribe to a publisher - the feed should say updates are available
8. Subscribe to a channel - the feed should say updates are available
